### PR TITLE
Tweaks

### PR DESCRIPTION
**Pull request by manolama**

### CODE REVIEW

File: core/src/main/java/net/opentsdb/aura/metrics/core/TimeSeriesShard.java [hunk 1]
```diff
@@ -584,13 +610,16 @@ public void run() {
...
       LOGGER.info(
-          "Purged {} segments {} timeseries {} meta records for shardId {}",
+          "Purged {} segments {} timeseries {} meta records for shardId {} in {} runs for {} seconds",
           segmentPurgeCount,
           tsPurgeCount,
           metaPurgeCount,
-          shardId);
+          shardId,
+          runs,
+          ((double) timeTaken / 1_000_000_000D));
```
Comment (smrutilal2) [inline core/src/main/java/net/opentsdb/aura/metrics/core/TimeSeriesShard.java:619]: redundant casting `(double)` while dr is already a double (1_000_000_000`D`)

File: core/src/main/java/net/opentsdb/aura/metrics/core/TimeSeriesShard.java [hunk 2]
```diff
@@ -606,7 +635,22 @@ public Job createNext() {
...
+      if (tsTableClone != null) {
+        if (!isPurgeDone.compareAndSet(false, true)) {
+          if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("Purge was already marked finished?",
+                    new RuntimeException("Call stack"));
+          }
+        } else {
+          tsTableClone.close();
+          tsTableClone = null;
+          new RuntimeException().printStackTrace();
```
Comment (smrutilal2) [inline core/src/main/java/net/opentsdb/aura/metrics/core/TimeSeriesShard.java:647]: ??

Comment (smrutilal2) [reply core/src/main/java/net/opentsdb/aura/metrics/core/TimeSeriesShard.java:647]: what is `new RuntimeException().printStackTrace()` for

File: core/src/main/java/net/opentsdb/aura/metrics/core/TimeSeriesShardIF.java [hunk 3]
```diff
@@ -59,4 +60,5 @@
 
   void flush();
 
+  MetricRegistry metricRegistry();
```
Comment (smrutilal2) [inline core/src/main/java/net/opentsdb/aura/metrics/core/TimeSeriesShardIF.java:63]: We need not expose metric registry from the Shard. Instead we can pass that to the `NewDocstore` as a constructor argument.

File: core/src/main/java/net/opentsdb/aura/metrics/meta/NewDocStore.java [hunk 4]
```diff
@@ -137,9 +173,11 @@ public void add(final String[] tagKeys, final String[] tagValues, long key) {
     final int index;
     if (resets.isEmpty()) {
       index = currentIndex++;
+      m_maxDocId.set(index, tags);
     } else {
       index = resets.first();
       resets.remove(index);
+      m_freeDocs.set(resets.getLongCardinality(), tags);
```
Comment (smrutilal2) [inline core/src/main/java/net/opentsdb/aura/metrics/meta/NewDocStore.java:180]: In theory, these metrics can have sparse data if no doc is added.

File: core/src/main/java/net/opentsdb/aura/metrics/meta/NewDocStore.java [hunk 5]
```diff
@@ -164,17 +203,29 @@ public void add(final String[] tagKeys, final String[] tagValues, long key) {
...
+              });
 
       RoaringBitmap rr = valueMap.get(tagValue);
       if (rr == null) {
-        valueMap.put(tagValue, RoaringBitmap.bitmapOf(index));
+        rr = RoaringBitmap.bitmapOf(index);
+        valueMap.put(tagValue, rr);
+        this.tagValues++;
+        tagValueBytes += tagValue.length();
+        rrSizeBytes += rr.getSizeInBytes();
```
Comment (smrutilal2) [inline core/src/main/java/net/opentsdb/aura/metrics/meta/NewDocStore.java:218]: Need to be careful about the instrumentation overhead tough.

File: core/src/main/java/net/opentsdb/aura/metrics/core/TimeSeriesShard.java (hunk 2 above)
Comment (manolama) [reply core/src/main/java/net/opentsdb/aura/metrics/core/TimeSeriesShard.java:647]: doh.... debugging flub.

File: core/src/main/java/net/opentsdb/aura/metrics/core/TimeSeriesShardIF.java (hunk 3 above)
Comment (manolama) [reply core/src/main/java/net/opentsdb/aura/metrics/core/TimeSeriesShardIF.java:63]: yeah but then we need to give it to the factory. I want to do that as part of a plugin refactoring I think with the TSDB code base.

File: core/src/main/java/net/opentsdb/aura/metrics/meta/NewDocStore.java (hunk 4 above)
Comment (manolama) [reply core/src/main/java/net/opentsdb/aura/metrics/meta/NewDocStore.java:180]: Yup :)

**Review by smrutilal2 — APPROVED** [review]

**Pull request merged**